src/msg/async/EventKqueue.cc

```diff
@@ -99,22 +211,66 @@ int KqueueDriver::event_wait(vector<FiredFileEvent> &fired_events, struct timeva
   int retval, numevents = 0;
   struct timespec timeout;
 
+  ldout(cct,10) << __func__ << " kqfd = " << kqfd << dendl;
+  if (!pthread_equal(mythread, pthread_self())) {
+    ldout(cct,10) << __func__ << " We changed thread from " << mythread << " to " << pthread_self() << dendl;
+    mythread = pthread_self();
+    kqfd = -1;
+  } else if ((kqfd != -1) && (test_kqfd() < 0)) {
```

Comment:
@yuyuyu101 
I have seen cases where it triggered an "invalid" kqfd. But perhaps that was during testing and the code still had bugs. Lets keep this code in for now. And in due time if we full understand what is going on, perhaps change it into an 'assert()' if this actually should not happen. But now I'm not 100% certain.